### PR TITLE
Fixed #7622 - Make the code:coverage Robo command work outside of CI

### LINKED ISSUE #7622: Make the code:coverage Robo command work outside of CI
#### Issue
This is part of #7344, but I'm splitting it out into a separate issue.

Right now the code coverage Robo command won't work outside of CI, which is a pretty unnecessary limitation. We should make sure it can be run locally as well.

### PR DESCRIPTION
**Pull request by Dillon-Brown**
## Description

Removes an unncessary limitation of the robo code coverage command so that it now allows a non-ci envionment get code coverage.

## Motivation and Context

Issue reference: #7622 

## How To Test This

1. Check Travis.
2. Run ```./vendor/bin/robo code:coverage``` and retrieve code covage info.
3. Run ```./vendor/bin/robo code:coverage --ci``` and it should fail with no ci detected.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [X] My code follows the code style of this project found [here](https://docs.suitecrm.com/community/contributing-code/coding-standards/).
- [ ] My change requires a change to the documentation.
- [X] I have read the [**How to Contribute**](https://docs.suitecrm.com/community/contributing-code/) guidelines.

### CODE REVIEW

**Comment by connorshea** [comment]
It's kind of weird to me that it doesn't output anything when the tests fail, but other than that it seems to work 🤷‍♂

File: lib/Robo/Plugin/Commands/CodeCoverageCommands.php [hunk 1]
```diff
@@ -50,18 +50,21 @@ class CodeCoverageCommands extends \Robo\Tasks
...
         // Get environment
-        if ($this->isEnvironmentTravisCI()) {
-            $range = $this->getCommitRangeForTravisCi();
-        } else {
-            throw new \RuntimeException('unable to detect continuous integration environment');
+        if ($opts['ci'] === true) {
+            if ($this->isEnvironmentTravisCI()) {
+                $range = $this->getCommitRangeForTravisCi();
+            } else {
+                throw new \RuntimeException('unable to detect continuous integration environment');
```
Comment (connorshea) [inline lib/Robo/Plugin/Commands/CodeCoverageCommands.php]:
```suggestion
                throw new \RuntimeException('Unable to detect continuous integration environment');
```

File: lib/Robo/Plugin/Commands/CodeCoverageCommands.php [hunk 2]
```diff
@@ -50,18 +50,21 @@ class CodeCoverageCommands extends \Robo\Tasks
     use RoboTrait;
 
     /**
-     * Runs code coverage for travis ci
+     * Runs code coverage
+     * @param array $opts
```
Comment (connorshea) [inline lib/Robo/Plugin/Commands/CodeCoverageCommands.php]: Should document the `@option` here

File: lib/Robo/Plugin/Commands/CodeCoverageCommands.php [hunk 3]
```diff
@@ -50,18 +51,21 @@ class CodeCoverageCommands extends \Robo\Tasks
     use RoboTrait;
 
     /**
-     * Runs code coverage for travis ci
+     * Runs code coverage
+     * @param array $opts optional command line argument if using a continuous integration environment
```
Comment (connorshea) [inline lib/Robo/Plugin/Commands/CodeCoverageCommands.php]: To be clear I meant something more like this:
```php
/**
 * Runs code coverage
 * @param array $opts
 * @option bool $ci Should be set to true if using a Continuous Integration environment.
 */
```

This is shown in the `Help` section of the Getting Started guide: https://robo.li/getting-started/

**Comment by Mac-Rae** [comment]
Assessed :+1:

**Pull request merged**